Import JSON parentJointName value

thoys opened the pull request:
## QA test
https://highfidelity.testrail.net/index.php?/cases/view/5149

marko8904 commented:
build this please

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5049/HighFidelity-Beta-PR11845-3c37bf8acda8eec6e3cddd103dea7c69abbed44a.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5049/HighFidelity-Beta-PR11845-3c37bf8acda8eec6e3cddd103dea7c69abbed44a.dmg

ZappoMan commented:
I think there's some merit to this idea, but I don't think we want to actually create "true" properties for joint names. Since the intent of this is to support import/export of client only entities that correctly attach to the named joints... the better approach would be to put all of this code in the `EntityItemProperties` and simply add an alias for `jointName` there. We can discuss this by voice if you need more guidance on how to do this.

thoys commented:
Oh read this just now, I made the following changes, let me know if you want me to move them elsewhere @ZappoMan

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5091/HighFidelity-Beta-PR11845-e94199acad2b7fe3ad8cf5bc0f24e5c7e5551d78.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5091/HighFidelity-Beta-PR11845-e94199acad2b7fe3ad8cf5bc0f24e5c7e5551d78.dmg

sethalves commented on libraries/entities/src/EntityTree.h:
```diff
@@ -278,6 +278,8 @@ class EntityTree : public Octree, public SpatialParentTree {
     QByteArray computeEncryptedNonce(const QString& certID, const QString ownerKey);
     bool verifyDecryptedNonce(const QString& certID, const QString& decryptedNonce, EntityItemID& id);
 
+    void setMyAvatar(AvatarData* myAvatar) { _myAvatar = myAvatar; }
```
Could you make this a smart-pointer rather than a bare one?  It's probably fine as it is, but I worry about introducing new crashes-on-shutdown.

ZappoMan commented:
@thoys you need a test plan... so this can get tested.

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5097/HighFidelity-Beta-PR11845-c3754d636902cecc55c6f5c6651ec99edca1ca7c.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5097/HighFidelity-Beta-PR11845-c3754d636902cecc55c6f5c6651ec99edca1ca7c.dmg

thoys commented:
This only works for importing. Export mechanism coming up..

thoys commented:
@sethalves @ZappoMan , I've added a way to export the parentJointName. Confirmed and works with the marketplace packaging app that @misslivirose made.

Also updated the TestRail

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5104/HighFidelity-Beta-PR11845-627fee80ec5b38f1a11b6ae5d2442890ba2af06d.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5104/HighFidelity-Beta-PR11845-627fee80ec5b38f1a11b6ae5d2442890ba2af06d.dmg

Nex-Pro commented:
Testing

Nex-Pro commented:
### PASS
result @ https://highfidelity.testrail.net/index.php?/tests/view/16637&group_by=cases:section_id&group_order=asc&group_id=323

thoys commented:
@sethalves it still says a review is required
